### PR TITLE
hv: mod: do not use explicit arch name when including headers

### PR DESCRIPTION
**Pull request by ylian16**
Instead of "#include <x86/foo.h>", use "#include <asm/foo.h>".

In other words, we are adopting the same practice in Linux kernel.

Tracked-On: #5975
Signed-off-by: Liang Yi <yi.liang@intel.com>
Reviewed-by: Jason Chen CJ <jason.cj.chen@intel.com>

### CODE REVIEW

**Review by dongyaozu — APPROVED** [review]

**Comment by wenlingz** [comment]
@ylian16 build fail, please check:
hypervisor/5976-1619671350/build//hypervisor/dm/vpci/ivshmem.o -MMD -MT /home/meta-acrn/workspace/build/master-acrn-sos/work/intel_corei7_64-oe-linux/acrn-hypervisor/5976-1619671350/build//hypervisor/dm/vpci/ivshmem.o
| dm/vpci/ivshmem.c:8:10: fatal error: x86/guest/vm.h: No such file or directory
|     8 | #include <x86/guest/vm.h>
|       |          ^~~~~~~~~~~~~~~~
| compilation terminated.
| Makefile:498: recipe for target '/home/meta-acrn/workspace/build/master-acrn-sos/work/intel_corei7_64-oe-linux/acrn-hypervisor/5976-1619671350/build//hypervisor/dm/vpci/ivshmem.o' failed
| make: *** [/home/meta-acrn/workspace/build/master-acrn-sos/work/intel_corei7_64-oe-linux/acrn-hypervisor/5976-1619671350/build//hypervisor/dm/vpci/ivshmem.o] Error 1
| make: Leaving directory '/home/meta-acrn/workspace/build/master-acrn-sos/work/intel_corei7_64-oe-linux/acrn-hypervisor/5976-1619671350/git/hypervisor'
| WARNING: /home/meta-acrn/workspace/build/master-acrn-sos/work/intel_corei7_64-oe-linux/acrn-hypervisor/5976-1619671350/temp/run.do_compile.1314:1 exit 1 from 'exit 1'
| 
NOTE: recipe acrn-hypervisor-5976-1619671350: task do_compile: Failed
ERROR: Task (/home/meta-acrn/workspace/meta-acrn/recipes-core/acrn/acrn-hypervisor.bb:do_compile) failed with exit code '1'
NOTE: recipe acrn-devicemodel-5976-1619671350: task do_packagedata: Started

**Comment by ylian16** [comment]
@wenlingz I have fixed the build error. Should be okay now.

**Comment by wenlingz** [comment]
@ylian16 please follow nanlin's comments, and add JIRA ID in git issue, thanks
NanlinXie commented 2 days ago
@ylian16 , please use #5920 as tracked on

**Review by dbkinder — APPROVED** [review]
doxygen changes LGTM

**Comment by dbkinder** [comment]
@ylian16 @yuchuyang Can you two talk about use of explicit arch names added in PR #5981 that are removed in this #5976 PR

File: misc/config_tools/board_config/board_c.py [hunk 1]
```diff
@@ -15,9 +15,9 @@ class RDT(enum.Enum):
     MBA = 2
 
 INCLUDE_HEADER = """
-#include <x86/board.h>
-#include <x86/vtd.h>
-#include <x86/msr.h>
+#include <asm/board.h>
+#include <asm/vtd.h>
+#include <asm/msr.h>
 #include <pci.h>
```
Comment (yuchuyang) [inline misc/config_tools/board_config/board_c.py:21]: Do you regenerate the files under  misc\config_tools\data\generic_board\generic_code\boards to reflect your changes of config_tools?

File: misc/config_tools/scenario_config/vm_configurations_c.py [hunk 2]
```diff
@@ -9,9 +9,9 @@
 import scenario_cfg_lib
 
 C_HEADER = scenario_cfg_lib.HEADER_LICENSE + r"""
-#include <x86/vm_config.h>
+#include <asm/vm_config.h>
 #include <vuart.h>
-#include <x86/pci_dev.h>
+#include <asm/pci_dev.h>
```
Comment (yuchuyang) [inline misc/config_tools/scenario_config/vm_configurations_c.py:14]: Do you regenerate the files under  misc\config_tools\data\generic_board\generic_code\<scenario_name> to reflect your changes of config_tools?

File: misc/config_tools/scenario_config/pt_intx_c.py [hunk 3]
```diff
@@ -17,7 +17,7 @@ def generate_file(vm_info, config):
 
     print("{}".format(scenario_cfg_lib.HEADER_LICENSE), file=config)
     print("", file=config)
-    print("#include <x86/vm_config.h>", file=config)
+    print("#include <asm/vm_config.h>", file=config)
```
Comment (yuchuyang) [inline misc/config_tools/scenario_config/pt_intx_c.py:20]: Do you regenerate the files under  misc\config_tools\data\generic_board\generic_code\<scenario_name> to reflect your changes of config_tools?

File: misc/config_tools/scenario_config/pci_dev_c.py [hunk 4]
```diff
@@ -131,12 +131,12 @@ def generate_file(vm_info, config):
...
     print("", file=config)
-    print("#include <x86/vm_config.h>", file=config)
+    print("#include <asm/vm_config.h>", file=config)
     print("#include <pci_devices.h>", file=config)
     print("#include <vpci.h>", file=config)
     print("#include <vbar_base.h>", file=config)
-    print("#include <x86/mmu.h>", file=config)
-    print("#include <x86/page.h>", file=config)
+    print("#include <asm/mmu.h>", file=config)
+    print("#include <asm/page.h>", file=config)
```
Comment (yuchuyang) [inline misc/config_tools/scenario_config/pci_dev_c.py:139]: Do you regenerate the files under  misc\config_tools\data\generic_board\generic_code\<scenario_name> to reflect your changes of config_tools?

File: misc/config_tools/scenario_config/ivshmem_cfg_h.py [hunk 5]
```diff
@@ -95,7 +95,7 @@ def generate_file(scenario_items, config):
 
     if vm_info.shmem.shmem_enabled == 'y':
         print("#include <ivshmem.h>", file=config)
-        print("#include <x86/pgtable.h>", file=config)
+        print("#include <asm/pgtable.h>", file=config)
```
Comment (yuchuyang) [inline misc/config_tools/scenario_config/ivshmem_cfg_h.py:98]: Do you regenerate the files under  misc\config_tools\data\generic_board\generic_code\<scenario_name> to reflect your changes of config_tools?

**Comment by ylian16** [comment]
No, this patch did not touch anything under misc/config_tools/data/.

**Comment by ylian16** [comment]
@Kinder, David ***@***.***>
Hi David, PR 5976 is about avoiding use of explicit arch name, i.e., “x86” when include arch specific header files. The Makefile will make sure that the correct include directory is in use so that there is no need for conditional compilation directives in the source code.

I am not aware of PR 5981 but if the generated files include arch specific headers it should follow the same approach. Relevant python generators under misc/config_tools have been updated accordingly in PR 5976 as well.

**Comment by yuchuyang** [comment]
@ylian16  could you please update the files under generated codes in order to reflect your change?

**Pull request merged**